### PR TITLE
[Extension] Dependency injection resolver and fix meshsync event subscription resolver

### LINKED ISSUE #4452: [UI] GraphQL subscription gives out undefined
#### Current Behavior

When loading the main dashboard, graphql subscription for control and dataplane states can be established. Data is transferred and it can confirmed by looking into the Network tab in Developer Tools in Chrome (or anything similar in other browsers) and even in GraphQL playground, data is successfully fetched. However, in the Relay graphql subscription, data is undefined.

#### Desired Behavior

Data is not undefined and able to update the states real time

#### Screenshots/Logs

https://user-images.githubusercontent.com/54849895/144454615-986d7277-32ff-4cbc-8123-cefafaf6dbbf.mp4

#### Environment

- Browser: Chrome 
- Host OS: Windows (WSL)
- Meshery Server Version: edge-latest
- Meshery Client Version: edge-latest
- Platform: Docker

---

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources

- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by bariqhibat**
**Description**

This PR fixes #4452

**Notes for Reviewers**
Linked issue: https://github.com/layer5labs/meshery-extensions/pull/477

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: internal/graphql/resolver/meshsync.go [hunk 1]
```diff
@@ -11,27 +13,74 @@ import (
 )
 
 var (
-	meshsyncYaml = "https://raw.githubusercontent.com/layer5io/meshery-operator/master/config/samples/meshery_v1alpha1_meshsync.yaml"
+	meshsyncYaml              = "https://raw.githubusercontent.com/layer5io/meshery-operator/master/config/samples/meshery_v1alpha1_meshsync.yaml"
```
Comment (leecalcote) [inline internal/graphql/resolver/meshsync.go:16]: @DelusionalOptimist, we want to stay away from manifests and stick with Helm charts, yes? Also, there is a MeshKit function that we should be reusing here, yeah?

Comment (DelusionalOptimist) [reply internal/graphql/resolver/meshsync.go:16]: Yes @leecalcote, the upstream code isn't using this. Its just a left over which I missed while cleaning up other references to manifests.
Also, as far as I can tell, it isn't being used even in this PR, right @bariqhibat? In that case, I think we should remove this :+1:

**Review by leecalcote — APPROVED** [review]

File: internal/graphql/resolver/meshsync.go (hunk 1 above)
Comment (bariqhibat) [reply internal/graphql/resolver/meshsync.go:16]: Yeah, this is unused. I'll delete this

**Review by sudo-NithishKarthik — APPROVED** [review]
LGTM . Thank you @bariqhibat

**Pull request merged**